### PR TITLE
Blog app - add forms

### PR DESCRIPTION
**Pull request by 78fatemanazari**
### Project requirements
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
It will return the first user from the database.
Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
- Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
+    respond_to do |format|
+      format.html do
+        if @post.save
+          flash[:success] = ' Post created '
+          redirect_to user_post_path(@user, @post)
+        else
+          render :new, status: :unprocessable_entity
+        end
+      end
+    end
```
Comment (V-Blaze) [inline app/controllers/posts_controller.rb]: - You have done great so far :+1: But there are a few requirements you still need to meet before moving to the next milestone. :+1: :100: 

- [ ] While creating a new post and I click on the submit button, even though the post seems to be created, this error got returned to be in the console and on the browser. The error seems to be referring to your view files, I'll suggest you fix this. :+1: :100: 

- - Screenshot reference:
Screenshot from 2023-11-30 21-50-46

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,29 +1,44 @@
...
+    @posts = Post.where(author_id: @user.id)
   end
 
   def show
     @post = Post.find(params[:id])
+    @current_user = current_user
     @author = User.find(@post.author_id)
+    @comments = @post.comments
+    @comment = Comment.new(user: @current_user, post: @post)
   end
```
Comment (V-Blaze) [inline app/controllers/posts_controller.rb]: - [ ] When I click on a post in order to go to the `post show page` I get this error in the browser, I suggest you also fix this. :+1: :100: 

- - Screenshot reference:
Screenshot from 2023-11-30 21-51-47

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def attr_user_post
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+  end
```
Comment (V-Blaze) [inline app/controllers/comments_controller.rb]: - - Requirement screenshot:
Selection_051


- [ ] According to the project requirement, the comment should be created onbehalf of the `current user` which you get by calling the `current_user` method you created. currently, you are getting the user by the ID passed from the request, although this works, that's not what was required. I suggest you stick to the requirement and use the `current_user` method here. :smile: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @78fatemanazari 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Created all the required controller actions ✔
-  Added routes correctly 👋 
-  Correct git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @78fatemanazari 👏 👏 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Assigned the `current_user` to the user attribute of a `Comment` instance. ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**